X-Account-Key: account2
X-UIDL: 1169648826.15570.mrelay3-1
X-Mozilla-Status: 0001
X-Mozilla-Status2: 00000000
Return-Path: <sha@pengutronix.de>
Delivered-To: online.fr-julien.boibessot@free.fr
Received: (qmail 15163 invoked from network); 24 Jan 2007 14:27:04 -0000
Received: from 83.236.181.26 (HELO metis.extern.pengutronix.de) (83.236.181.26)
  by mrelay3-1.free.fr with SMTP; 24 Jan 2007 14:27:04 -0000
Received: from leda.ptxnet.pengutronix.de ([10.1.0.71])
	by metis.extern.pengutronix.de with esmtp (Exim 3.36 #1 (Debian))
	id 1H9iZZ-0002mz-00; Wed, 24 Jan 2007 14:53:57 +0100
Received: from sha by leda.ptxnet.pengutronix.de with local (Exim 4.63)
	(envelope-from <sha@localhost.localdomain>)
	id 1H9iZZ-0001hB-ED; Wed, 24 Jan 2007 14:53:57 +0100
Date: Wed, 24 Jan 2007 14:53:57 +0100
From: Sascha Hauer <s.hauer@pengutronix.de>
To: julien.boibessot@free.fr
Cc: linux-arm-kernel@lists.arm.linux.org.uk
Subject: [PATCH] move imxfb.h into c file
Message-ID: <20070124135357.GC5034@localhost.localdomain>
Mail-Followup-To: julien.boibessot@free.fr,
	linux-arm-kernel@lists.arm.linux.org.uk
References: <1169639047.45b74687c1102@imp.free.fr>
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
In-Reply-To: <1169639047.45b74687c1102@imp.free.fr>
User-Agent: Mutt/1.5.13 (2006-08-11)

This patch moves the content of drivers/video/imxfb.h into
drivers/video/imxfb.c since it is only used in this one file

Signed-Off-by: Sascha Hauer <s.hauer@pengutronix.de>

---
 drivers/video/imxfb.c |   59 +++++++++++++++++++++++++++++++++++++++-
 drivers/video/imxfb.h |   73 --------------------------------------------------
 2 files changed, 58 insertions(+), 74 deletions(-)

Index: drivers/video/imxfb.c
===================================================================
--- a/drivers/video/imxfb.c.orig
+++ b/drivers/video/imxfb.c
@@ -41,7 +41,64 @@
  */
 #define DEBUG_VAR 1
 
-#include "imxfb.h"
+/*
+ * These are the bitfields for each
+ * display depth that we support.
+ */
+struct imxfb_rgb {
+	struct fb_bitfield	red;
+	struct fb_bitfield	green;
+	struct fb_bitfield	blue;
+	struct fb_bitfield	transp;
+};
+
+#define RGB_16	(0)
+#define RGB_8	(1)
+#define NR_RGB	2
+
+struct imxfb_info {
+	struct device		*dev;
+	struct imxfb_rgb	*rgb[NR_RGB];
+
+	u_int			max_bpp;
+	u_int			max_xres;
+	u_int			max_yres;
+
+	/*
+	 * These are the addresses we mapped
+	 * the framebuffer memory region to.
+	 */
+	dma_addr_t		map_dma;
+	u_char *		map_cpu;
+	u_int			map_size;
+
+	u_char *		screen_cpu;
+	dma_addr_t		screen_dma;
+	u_int			palette_size;
+
+	dma_addr_t		dbar1;
+	dma_addr_t		dbar2;
+
+	u_int			pcr;
+	u_int			pwmr;
+	u_int			lscr1;
+	u_int			dmacr;
+	u_int			cmap_inverse:1,
+				cmap_static:1,
+				unused:30;
+
+	void (*lcd_power)(int);
+	void (*backlight_power)(int);
+};
+
+#define IMX_NAME	"IMX"
+
+/*
+ * Minimum X and Y resolutions
+ */
+#define MIN_XRES	64
+#define MIN_YRES	64
+
 
 static struct imxfb_rgb def_rgb_16 = {
 	.red	= { .offset = 8,  .length = 4, },
Index: drivers/video/imxfb.h
===================================================================
--- a/drivers/video/imxfb.h
+++ b//dev/null
@@ -1,73 +0,0 @@
-/*
- * linux/drivers/video/imxfb.h
- *
- *  Freescale i.MX Frame Buffer device driver
- *
- *  Copyright (C) 2004 S.Hauer, Pengutronix
- *
- *  Copyright (C) 1999 Eric A. Thomas
- *   Based on acornfb.c Copyright (C) Russell King.
- *
- * This file is subject to the terms and conditions of the GNU General Public
- * License.  See the file COPYING in the main directory of this archive
- * for more details.
- */
-
-/*
- * These are the bitfields for each
- * display depth that we support.
- */
-struct imxfb_rgb {
-	struct fb_bitfield	red;
-	struct fb_bitfield	green;
-	struct fb_bitfield	blue;
-	struct fb_bitfield	transp;
-};
-
-#define RGB_16	(0)
-#define RGB_8	(1)
-#define NR_RGB	2
-
-struct imxfb_info {
-	struct device		*dev;
-	struct imxfb_rgb	*rgb[NR_RGB];
-
-	u_int			max_bpp;
-	u_int			max_xres;
-	u_int			max_yres;
-
-	/*
-	 * These are the addresses we mapped
-	 * the framebuffer memory region to.
-	 */
-	dma_addr_t		map_dma;
-	u_char *		map_cpu;
-	u_int			map_size;
-
-	u_char *		screen_cpu;
-	dma_addr_t		screen_dma;
-	u_int			palette_size;
-
-	dma_addr_t		dbar1;
-	dma_addr_t		dbar2;
-
-	u_int			pcr;
-	u_int			pwmr;
-	u_int			lscr1;
-	u_int			dmacr;
-	u_int			cmap_inverse:1,
-				cmap_static:1,
-				unused:30;
-
-	void (*lcd_power)(int);
-	void (*backlight_power)(int);
-};
-
-#define IMX_NAME	"IMX"
-
-/*
- * Minimum X and Y resolutions
- */
-#define MIN_XRES	64
-#define MIN_YRES	64
-


-- 
 Dipl.-Ing. Sascha Hauer | http://www.pengutronix.de
  Pengutronix - Linux Solutions for Science and Industry
    Handelsregister: Amtsgericht Hildesheim, HRA 2686
      Hannoversche Str. 2, 31134 Hildesheim, Germany
    Phone: +49-5121-206917-0 |  Fax: +49-5121-206917-9


